src/analysis.py

```diff
@@ -0,0 +1,219 @@
+        object return by `cross_validate`
+    results_dict: dict
+        dictionary to store results
+
+    Returns
+    ----------
+        None
+
+    """
+    results[model] = {
+        "mean_validation_accuracy": "{:0.4f}".format(np.mean(scores["test_score"])),
+        "mean_train_accuracy": "{:0.4f}".format(np.mean(scores["train_score"])),
+        "mean_fit_time (s)": "{:0.4f}".format(np.mean(scores["fit_time"])),
+        "mean_score_time (s)": "{:0.4f}".format(np.mean(scores["score_time"])),
+        "std_train_score": "{:0.4f}".format(scores["train_score"].std()),
+        "std_test_score": "{:0.4f}".format(scores["test_score"].std()),
+    }
+
+
+
+def train_model(dataframe):
+    """
+    Processes the data, trains the model, and returns a dataframe showing the results
+
+    Parameters
+    ----------
+    dataframe : pd.Dataframe
+        Dataframe to be used in model
+
+    Returns
+    ----------
+        pandas.Dataframe
+
+    """
+    train_df = dataframe
+    X_train = train_df.drop(columns=['align'])
+    y_train = train_df['align']
+
+    numeric_features = ['appearances']
+    categorical_features = ['name', 'id', 'eye', 'hair', 'year', 'publisher', 'sex']
+    drop_features = ['gsm', 'first_appearance']
+
+    numeric_transformer = Pipeline(
+        steps=[
+            ("imputer", SimpleImputer(strategy='median')),
+            ("scaler", StandardScaler()),
+        ]
+    )
+
+    categorical_transformer = Pipeline(
+        steps=[
+            ("imputer", SimpleImputer(strategy='most_frequent')),
+            ("onehot", OneHotEncoder(handle_unknown='ignore'))
+        ]
+    )
+
+    preprocessor = make_column_transformer(
+        ("drop", drop_features),
+        (numeric_transformer, numeric_features),
+        (categorical_transformer, categorical_features),
+    )
+
+    results_df = {}
+
+    models = {
+        "Dummy Classifier": DummyClassifier(),
+        "Decision Tree Classifier": DecisionTreeClassifier(),
+        "Random Forest Classifier": RandomForestClassifier(),
+        "SVM": SVC(),
+        "OVR LogisticRegression": OneVsRestClassifier(LogisticRegression()),
+        "LogisticRegression": LogisticRegression(),
+    }
+
+    for key, value in models.items():
+        pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor),
+               (key, value)
+                ]
+            )
+        scores = cross_validate(pipeline, X_train, y_train, return_train_score=True, n_jobs=-1, verbose=1)
+        store_cross_val_results(key, scores, results_df)
+
+    results = pd.DataFrame(results_df)
+    results.reset_index(inplace=True)
+    results = results.rename(columns = {'index':'Scores'})
+    return results
+
+
+
+def save_img(data_frame, output_folder, file_name):
+    fig_1, ax_1 = render_table(data_frame, header_columns=0, col_width=5)
+    fig_1.savefig(output_folder + "/" + file_name)
+
+def main(input_file_path, output_folder_path):
+    print(input_file_path)
```

Comment:
Print statements and commented out code cleanup

Reply:
Do you want me to remove print statements?